pkg/tfgen/examples_coverage_exporter.go

```diff
@@ -0,0 +1,158 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file implements the methods used by the Coverage Tracker in order
+// to export the data it collected into various JSON formats.
+
+package tfgen
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// The export utility's main structure, where it stores the desired output directory
+// and a reference to the CoverageTracker that created it
+type CoverageExportUtil struct {
+	CT *CoverageTracker // Reference to the Coverage Tracker that wants to turn its data into a file
+}
+
+func newCoverageExportUtil(coverageTracker *CoverageTracker) CoverageExportUtil {
+	return CoverageExportUtil{coverageTracker}
+}
+
+// The entire export utility interface. Will attempt to export the Coverage Tracker's data into the
+// specified output directory, and will panic if an error is encountered along the way
+func (CE *CoverageExportUtil) tryExport(outputDirectory string) {
+	CE.exportUploadableResults(outputDirectory, "summary.json")
+	CE.exportSummarizedResults(outputDirectory, "concise.json")
+}
+
+// Three different ways to export coverage data:
+// The first mode, using a large provider > example map
+func (CE *CoverageExportUtil) exportFullResults(outputDirectory string, fileName string) {
+
+	// The Coverage Tracker data structure remains identical, the only thing added in the file is the name of the provider
+	ProviderNameToExamplesMap := map[string]map[string]GeneralExampleInfo{CE.CT.ProviderName: CE.CT.EncounteredExamples}
+
+	jsonOutputLocation := createJsonOutputLocation(outputDirectory, fileName)
+	marshalAndWriteJson(ProviderNameToExamplesMap, jsonOutputLocation)
+}
+
+// The second mode, similar to existing Pulumi coverage Json files uploadable to redshift
+func (CE *CoverageExportUtil) exportUploadableResults(outputDirectory string, fileName string) {
+
+	// The Coverage Tracker data structure is flattened down to the example level, and they all
+	// get individually written to the file in order to not have the "{ }" brackets at the start and end
+	type SingleExampleResult struct {
+		ProviderName    string
+		ProviderVersion string
+		ExampleName     string
+		FailedLanguages []LanguageConversionResult
+	}
+
+	jsonOutputLocation := createJsonOutputLocation(outputDirectory, fileName)
+
+	// All the examples in the map are iterated by key and marshalled into one large byte array
+	// separated by \n, making the end result look like a bunch of Json files that got concatenated
+	var result = []byte{}
```

Comment:
You can simplify this to

```suggestion
	var result []byte
```